src/utils/logger.c

```diff
@@ -32,22 +32,19 @@ void logger_free()
     l->fd = -1;
 }
 
-int buffer_set_timestamp(char *buf)
+static size_t buffer_set_timestamp(char *buf)
 {
-    int len;
-    char *timestr;
+    size_t len;
     struct tm *local;
     time_t tm = time(NULL);
     local = localtime(&tm);
-    timestr = asctime(local);
-    len = snprintf(buf, LOG_BUFFER_SIZE, "%s", timestr);
-    if (len > LOG_BUFFER_SIZE)
-        len = LOG_BUFFER_SIZE;
-    buf[len - 1] = ' ';
+    len = strftime(buf, LOG_BUFFER_SIZE, "%a %b %e %T %Y ", local);
+    if (!len)
```

Comment:
personally not a fan of using ! with numbers, len is not a boolean value. I'd do `if(len == 0)`. not a big deal tho

Reply:
makes sense. amended. thx.